### PR TITLE
EES-2448 Only create ReleaseStatus row if ApprovalStatus changes

### PR DESCRIPTION
**Pull request by markhiveit**
The same release update endpoint is being used on three different pages: to change a release's ApprovalStatus, to add/edit PreRelease access list content, and to alter a release's summary info. 

By only adding a new ReleaseStatus row when the ApprovalStatus changes, we ensure that we only add a new ReleaseStatus entry when the user is on the Sign off page. This prevents the endpoint erroring if it is used to add/edit an access list or edit a release's summary info.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:328]: How come you set an id here now?

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:342]: I think an approver can change the release note when a Release is already scheduled.  I think this change might stop this or force them to unapprove first before reapproving with a different note so worth checking it.

I.e, they won't be able to do this, and the change would be silently ignored I think? 

image

Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:342]: We discussed this issue on standup and it was decided that a new endpoint would be created to update the release from the sign off page. See EES-2448 ticket on Jira.

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1141]: You can use:

```
var updatedRelease = result.AssertRight();
```

instead for brevity.

Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1141]: Then you also don't need to use `result.Right.` all the time below either

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1169]: Think the comment is incorrect?

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1264]: Remove this?

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1350]: In the same way we have `var xxx = result.AssertRight()`, it'd probably be nice to have an `result.AssertValidationProblems(prob1, prob2, prob3)` that does both the Left check, and also does 1 or more `AssertValidationProblem` checks as well.

Not suggesting as part of this PR though!

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1460]: Why did these need to change out of interest?

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs
Comment (duncan-at-hiveit) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:425]: Do we need these TODOs?

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1141]: I think this is only on the SOW branch at the moment?

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1460]: To prevent the bug when trying to create an prerelease access list from occurring, the UpdateRelease method still needs to only add a ReleaseStatus if the ApprovalStatus has changed - even though nothing using that endpoint will change the ApprovalStatus.

Once UpdateRelease is updated as part of the fixup ticket after this PR is merged, ReleaseUpdateViewModel will no longer take an ApprovalStatus and so this will no longer be an issue. So, to finally answer your question, I needed to do this to make the test pass. :)

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs
Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:425]: I've removed them

File: src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseServiceTests.cs:1141]: Oh sorry!

**Review by duncan-at-hiveit — APPROVED** [review]

**Pull request merged**
